### PR TITLE
Homework 1: Basics of Python and functional programming

### PR DESCRIPTION
**Pull request by conk3r353**
homework1.py - data types
homework2.py - python basic capabilities
homework3.py - python functions

### CODE REVIEW

File: home1/homework2.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import functools
+
+
+def analyze_students(data: typing.Dict) -> typing.Set:
+    return {(name, subject, functools.reduce(lambda x, y: x*y, data[name][subject])) for name in data for subject in data[name] if subject != '1C'}
+
+
+def validate_data(data: typing.Dict) -> bool:
+    for name in data:
+        if type(name) != str:
```
Comment (andrewrogow) [inline home1/homework2.py]: Better way is using `isinstance()` for validating types of values. It is more readable for another developers and as result more popular.

File: home1/homework3.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+class FibIterator:
+    def __init__(self):
+        self.a = 0
```
Comment (andrewrogow) [inline home1/homework3.py]: Use full names for variables.
variables named like `a,b,i,j,x,y` are not clear.
use short form for lambda functions or any  simple structures where name of variable doesn't make sense in their case.

File: home1/homework2.py (hunk 1 above)
Comment (conk3r353) [reply home1/homework2.py]: Fixed it.

File: home1/homework3.py (hunk 2 above)
Comment (conk3r353) [reply home1/homework3.py]: Also done.

**Pull request merged**